src/registry.ts

```diff
@@ -1,52 +1,45 @@
-import { pki } from "node-forge";
 import { AxiosResponse } from "axios";
 import { SkynetClient } from "./client";
-import { FileID, User } from "./skydb";
 import { defaultOptions, hexToUint8Array } from "./utils";
 import { Buffer } from "buffer";
+import { HashDataKey, PublicKey, Signature } from "./crypto";
 
 const defaultRegistryOptions = {
   ...defaultOptions("/skynet/registry"),
 };
 
-export type RegistryValue = {
-  tweak: Uint8Array;
+export type RegistryEntry = {
+  datakey: string;
   data: string;
   revision: number;
 };
 
-export type SignedRegistryValue = {
-  value: RegistryValue;
-  signature: pki.ed25519.NativeBuffer;
+export type SignedRegistryEntry = {
+  entry: RegistryEntry;
+  signature: Signature;
 };
 
-export async function lookupRegistry(
+export async function getEntry(
   this: SkynetClient,
-  user: User,
-  fileID: FileID,
+  publickey: PublicKey,
+  datakey: string,
   customOptions = {}
-): Promise<SignedRegistryValue | null> {
+): Promise<SignedRegistryEntry | null> {
   const opts = {
     ...defaultRegistryOptions,
     ...this.customOptions,
     ...customOptions,
   };
 
+  const userID = publickey.toString("hex");
```

Comment:
(nit) would not call this `userID` and simply inline:
`publickey: ed25519:${publickey.toString("hex")}`